src/main/java/com/xiaomi/infra/pegasus/rpc/async/ClusterManager.java

```diff
@@ -43,32 +43,25 @@
     logger.info("operating system name: {}", osName);
   }
 
-  public ClusterManager(
-      int timeout,
-      int io_threads,
-      boolean enableCounter,
-      String perfCounterTags,
-      int pushIntervalSecs,
-      String[] address_list)
-      throws IllegalArgumentException {
-    setTimeout(timeout);
-    this.enableCounter = enableCounter;
+  public ClusterManager(ClusterOptions opts) throws IllegalArgumentException {
+    setTimeout(opts.operationTimeout());
+    this.enableCounter = opts.enablePerfCounter();
     if (enableCounter) {
-      MetricsManager.detectHostAndInit(perfCounterTags, pushIntervalSecs);
+      MetricsManager.detectHostAndInit(opts.perfCounterTags(), opts.pushCounterIntervalSecs());
     }
 
     replicaSessions = new ConcurrentHashMap<rpc_address, ReplicaSession>();
-    replicaGroup = getEventLoopGroupInstance(io_threads);
+    replicaGroup = getEventLoopGroupInstance(opts.asyncWorkers());
     metaGroup = getEventLoopGroupInstance(1);
     tableGroup = getEventLoopGroupInstance(1);
 
-    metaList = address_list;
+    metaList = opts.metaList();
     // the constructor of meta session is depend on the replicaSessions,
     // so the replicaSessions should be initialized earlier
-    metaSession = new MetaSession(this, address_list, timeout, 10, metaGroup);
+    metaSession = new MetaSession(this, opts.metaList(), opts.operationTimeout(), 10, metaGroup);
```

Comment:
I think you can pass options to the constructor of MetaSession

Reply:
Yes but this is not my concern in this PR. We can do this refactoring later when it needs to add new param to MetaSession.

Reply:
Leave it later when it needs to add param to MetaSession.